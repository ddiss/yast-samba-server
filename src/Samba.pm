# Copyright 2004, Novell, Inc.  All rights reserved.

=head1 NAME

YaPI::Samba - Samba server configuration API

=head1 PREFACE

This package is the public Yast2 API to configure the Samba server.

=head1 SYNOPSIS

use YaPI::Samba

$serverRole = DetermineRole()

  returns a string representing the currently configure role or
  undef on failure.

$enabled = GetServiceStatus()

  returns true if the services smbd and nmbd are enabled, false if either of
  them is disabled or undef on failure.

$enabled = ModifyService($enable)

  enabled/disables the smbd and nmbd services. returns the status of the
  service, undef on failure.

$result = ModifyServerAsBDC($pdc_name)

  configures the samba server as a Backup Domain Controller with the given
  PDC. return undef on failure.

$result = ModifyServerAsPDC()

  configures the samba server as a Backup Domain Controller with the given
  PDC. return undef on failure.

$result = ModifyServerAsStandalone()

  configures the samba server as a Backup Domain Controller with the given
  PDC. return undef on failure.

$description = GetServerDescription()

  returns the server comment string, or undef it is not configured or on
  failure.

$result = ModifyServerDescription($description)

  sets a new server comment string. returns undef on failure

$passdbHash = GetSAMConfiguration($samString)

  returns a hash containing the configuration options related to a passdb
  backend. return undef on failure.

$result = ModifySAMConfiguration($samString,$passdbHash)

  sets the options for a passdb backend. returns undef on failure.

$result = SetDefaultSAM($samString)

  sets the given passdb backend as default (used for creating new users).
  returns undef on error.

$result = CreateSAMConfiguration($samString,$passdbHash)

  creates a new passdb backend using the given options. returns undef
  on error.

$result = RemoveSAM($samString)

  removes the given passdb backend. returns undef on failure.

$result = EnableShare($shareName,$enable)

  enables/disabled the given share. returns undef on failure.

$result = CreateShare($shareName,$options)

  creates a new share using the given options. returns undef on failure.

$result = RemoveShare($shareName)

  removes a share completely. returns undef on failure.

$result = ModifyShare($shareName,$options)

  modifies an existing share using the given options. returns undef on failure.

$options = GetShare($shareName)

  returns a hash describing the given share or undef on failure.

$shares = GetAllDirectories()

  returns a list of share names, which are configured as disk shares. returns
  undef on failure.

$result = EnableHomes($enable)

  enables/disables special share [homes]. returns undef on failure.

$result = EnableNetlogon($enable)

  enables/disables special share [netlogon]. returns undef on failure.

$shares = GetAllPrinters()

  returns a list of share names, which are configured as printers. returns
  undef on failure.

$result = EnablePrinters($printerList,$enable)

  enables/disables a list of printers. returns undef on failure.

=head1 DESCRIPTION

=over 2

=cut

package YaPI::Samba;
use YaST::YCP;
BEGIN { push( @INC, '/usr/share/YaST2/modules/' ); }

if(not defined do("YaPI.inc")) {
    die "'$!' Can not include YaPI.inc";
}

#######################################################

use strict;

#######################################################
# API start
#######################################################

=item *
C<$hostList = GetServiceStatus ();>

Returns the current status of smb and nmb services. True means 
the services are both started in at least on runlevel.
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{GetServiceStatus} = ["function", "string" ]; }
sub GetServiceStatus {
    # TODO:
    $self = shift;    
    return undef;
}

=item *
C<$serverRole = DetermineRole();>

This function determines role of a server in the SMB network. 
The return values can be unknown, standalone, bdc and pdc.
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{DetermineRole} = ["function", "string" ]; }
sub DetermineRole {
    # TODO:
    $self = shift;    
    return undef;
}

=item *
C<$enabled = ModifyService($enable);>

Modifies the status of the service. If the parameter is true, 
smb and nmb services are enabled in the default runlevels, 
if there were not enabled already in at least single runlevel. 
False will turn off the service in all runlevels.
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{ModifyService} = ["function", "boolean", "boolean" ]; }
sub ModifyService {
    # TODO:
    $self = shift;    
    $enable = shift;
    return undef;
}

=item *
C<$result = ModifyServerAsBDC($pdc_name)>

Configures the global settings of a server to behave like a 
backup domain controller. The primary domain controller is setup by the argument.
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{ModifyServerAsBDC} = ["function", "void", "string" ]; }
sub ModifyServerAsBDC {
    # TODO:
    $self = shift;   
    $pdc = shift; 
    return undef;
}

=item *
C<$result = ModifyServerAsPDC()>

Configures the global settings of a server to behave like a primary domain controller.
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{ModifyServerAsPDC} = ["function", "void" ]; }
sub ModifyServerAsPDC {
    # TODO:
    $self = shift;
    return undef;
}

=item *
C<$result = ModifyServerAsStandalone();>

Configures the global settings of a server to behave like a standalone 
server not taking part in any domain.
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{ModifyServerAsStandalone} = ["function", "void" ]; }
sub ModifyServerAsStandalone {
    # TODO:
    $self = shift;    
    return undef;
}

=item *
C<$description = GetServerDescription();>

Returns the configured description of the server. 
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{GetServerDescription} = ["function", "string" ]; }
sub GetServerDescription {
    # TODO:
    $self = shift;    
    return undef;
}

=item *
C<$result = ModifyServerDescription($description);>

Configures the description of the server shown in clients. 
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{ModifyServerDescription} = ["function", "void", "string" ]; }
sub ModifyServerDescription {
    # TODO:
    $self = shift;    
    $description = shift;
    return undef;
}

=item *
C<$passdbHash = GetSAMConfiguration($samString);>

Returns a list of options specified for the given SAM. The structure of the options is sam-type specific.
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{GetSAMConfiguration} = ["function", [ "map", "string", "any" ], "string" ]; }
sub GetSAMConfiguration {
    # TODO
    $self = shift;
    $sam = shift;
    
    return undef;
}

=item *
C<$result = ModifySAMConfiguration($samString, $passdbHash);>

Modifies the configuration of the given sam. The structure of the hash 
must follow the structure as specified for GetSAMConfiguration. 
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{ModifySAMConfiguration} = ["function", "boolean", "string", ["map", "string", "any" ] ]; }
sub ModifySAMConfiguration {
    # TODO
    $self = shift;
    $sam = shift;
    %options = shift;
    
    return undef;
}

=item *
C<$result = SetDefaultSAM($samString);>

Sets the SAM as default one, meaning that adding a new user will be done using this SAM. 
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{SetDefaultSAM} = ["function", "boolean", "string" ]; }
sub SetDefaultSAM {
    # TODO
    $self = shift;
    $sam = shift;
    
    return undef;
}

=item *
C<$result = CreateSAMConfiguration($samString, $passdbHash);>

Creates a new SAM using the given name and configuration. 
The structure of the hash must follow the structure as specified for GetSAMConfiguration.
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{CreateSAMConfiguration} = ["function", "boolean", "string", [ "map", "string", "any" ] , "boolean" ]; }
sub CreateSAMConfiguration {
    # TODO
    $self = shift;
    $sam = shift;
    %options = shift;
    $default = shift;
    
    return undef;
}

=item *
C<$result = RemoveSAM($samString);>

Removes the specified sam. It is not possible to remove the default one.
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{RemoveSAM} = ["function", "boolean", "string" ]; }
sub RemoveSAM {
    # TODO
    $self = shift;
    $sam = shift;
    
    return undef;
}

=item *
C<$result = EnableShare($shareName,$enable);>

Enables/disables the given share. 
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{EnableShare} = ["function", "boolean", "string", "boolean"] ; }
sub EnableShare {
    # TODO
    $self = shift;
    $name = shift;
    $on = shift;
    
    return undef;
}

=item *
C<$result = CreateShare($shareName,$options);>

Creates a new share with the given name and initial options. 
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{CreateShare} = ["function", "boolean", "string", [ "map", "string", "any" ] ] ; }
sub CreateShare {
    # TODO
    $self = shift;
    $name = shift;
    %options = shift;
    
    return undef;
}

=item *
C<$result = RemoveShare($shareName);>

Removes the given share. 
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{RemoveShare} = ["function", "boolean", "string"] ; }
sub RemoveShare {
    # TODO
    $self = shift;
    $name = shift;
    
    return undef;
}

=item *
C<$result = ModifyShare($shareName,$options);>

Modifies the given share to use the given options. 
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{ModifyShare} = ["function", "boolean", "string", [ "map", "string", "any" ] ] ; }
sub ModifyShare {
    # TODO
    $self = shift;
    $name = shift;
    %options = shift;
    
    return undef;
}

=item *
C<$options = GetShare($shareName);>

Returns a hash describing the given share.
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{ModifyShare} = ["function", "boolean", "string", [ "map", "string", "any" ] ] ; }
sub ModifyShare {
    # TODO
    $self = shift;
    $name = shift;
    %options = shift;
    
    return undef;
}

=item *
C<$shares = GetAllDirectories();>

Returns a list of all shares configured to provide a directory, 
including special-purpose shares like homes and netlogon.
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{GetAllDirectories} = ["function", [ "list", "string" ] ]; } 
sub GetAllDirectories {
    # TODO
    $self = shift;
    
    return undef;
}

=item *
C<$result = EnableHomes($enable);>

Enables a special-purpose share for sharing homes of a user. 
If the share does not exist, a default template is used.
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{EnableHomes} = ["function", "void",  "boolean"] ; }
sub EnableHomes {
    # TODO
    $self = shift;
    $on = shift;
    
    return undef;
}

=item *
C<$result = EnableNetlogon($enable);>

Enables a special-purpose share for login scripts. 
If the share does not exist, a default template is used.
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{EnableNetlogon} = ["function", "void",  "boolean"] ; }
sub EnableNetlogon {
    # TODO
    $self = shift;
    $on = shift;
    
    return undef;
}

=item *
C<$shares = GetAllPrinters();>

Returns a list of all printers configured to be shared.
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{GetAllPrinters} = ["function", [ "list", "string" ] ]; }
sub GetAllPrinters {
    # TODO
    $self = shift;
    
    return undef;
}

=item *
C<$result = EnablePrinters($printerList,$enable);>

Enables/disables sharing of the given printers. 
On error, undef is returned and the Error() function can be used
to get the error hash.

=cut

BEGIN { $TYPEINFO{EnablePrinters} = ["function", "boolean", [ "list", "string" ], "boolean"] ; }
sub EnablePrinters {
    # TODO
    $self = shift;
    @printer_names = shift;
    $enable = shift;
    
    return undef;
}

#######################################################
# API end
#######################################################

=back

=cut

42;
