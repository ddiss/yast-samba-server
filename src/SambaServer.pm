# File:		modules/SambaServer.ycp
# Package:	Configuration of samba-server
# Summary:	Data for configuration of samba-server, input and output functions.
# Authors:	Stanislav Visnovsky <visnov@suse.cz>
#		Martin Lazar <mlazar@suse.cz>
#
# $Id$
#
# Representation of the configuration of samba-server.
# Input and output routines.


package SambaServer;

use strict;
use Switch 'Perl6';
use Data::Dumper;

use YaST::YCP qw(:DATA :LOGGING);
use YaPI;

textdomain "samba-server";
our %TYPEINFO;

BEGIN {
YaST::YCP::Import("SCR");
YaST::YCP::Import("Mode");
YaST::YCP::Import("Report");
YaST::YCP::Import("Summary");
YaST::YCP::Import("Progress");
#HELPME: YaST::YCP::Import("Directory");
#HELPME: YaST::YCP::Import("SuSEFirewall");
YaST::YCP::Import("PackageSystem");

YaST::YCP::Import("SambaRole");
YaST::YCP::Import("SambaLDAP");
YaST::YCP::Import("SambaConfig");
YaST::YCP::Import("SambaService");
YaST::YCP::Import("SambaBackend");
YaST::YCP::Import("SambaSecrets");
YaST::YCP::Import("SambaNmbLookup");
}

use constant {
#HELPME:    DONE_ONCE_FILE => Directory->vardir . "/samba_server_done_once"
    DONE_ONCE_FILE => "/var/lib/YaST2" . "/samba_server_done_once"
};


my $Modified;

# list of required packages
my $RequiredPackages = ["samba", "samba-client"];

# Abort function
# return boolean return true if abort
#my $AbortFunction = undef;

my $GlobalsConfigured = 0;


#sub ServerReallyAbort {
#    my ($self) = @_;
#    return !GetModified() || Popup->ReallyAbort(Boolean(1));
#}

# Abort function
#sub Abort {
#    my ($self) = @_;
#    return defined $AbortFunction ? &$AbortFunction() : 0;
#}

# Set modify flag
BEGIN{ $TYPEINFO{SetModified} = ["function", "void"] }
sub SetModified {
    my ($self) = @_;
    $Modified = 1;
}

# Data was modified?
BEGIN{ $TYPEINFO{GetModified} = ["function", "boolean"] }
sub GetModified {
    my ($self) = @_;
    return $Modified 
	|| SambaConfig->GetModified() 
	|| SambaService->GetModified() 
	|| SambaBackend->GetModified()
	|| SambaTrustDom->GetModified();
};

# Read all samba-server settings
# @param force_reread force reread configuration
# @param no_progreass_bar disable progress bar
# @return true on success
BEGIN{ $TYPEINFO{Read} = ["function", "boolean"] }
sub Read {
    my ($self) = @_;

    # Samba-server read dialog caption
    my $caption = __("Initializing Samba Server Configuration");

    # We do not set help text here, because it was set outside
    Progress->New($caption, " ", 5, [
	    # translators: progress stage 1/5
	    __("Read global Samba settings"),
	    # translators: progress stage 4/5
	    __("Read the LDAP settings"),
	    # translators: progress stage 5/5
	    __("Read the firewall settings")
	], [
	    # translators: progress step 1/5
	    __("Reading global Samba settings..."),
	    # translators: progress step 4/5
	    __("Reading the LDAP settings..."),
	    # translators: progress step 5/5
	    __("Reading the firewall settings..."),
	    # translators: progress finished
	    __("Finished")
	],
	""
    );

    # read global settings
    Progress->NextStage();
    
    # check installed packages
    unless (Mode->test()) {
	PackageSystem->CheckAndInstallPackagesInteractive($RequiredPackages) or return 0;
    }

    SambaConfig->Read();
    SambaSecrets->Read();
    SambaService->Read();

    $GlobalsConfigured = $self->Configured();

    y2milestone("Service:". (SambaService->GetServiceAutoStart() ? "Enabled" : "Disabled"));
    y2milestone("Role:". SambaRole->GetRoleName());

    # start nmbstatus in background
    SambaNmbLookup->Start() unless Mode->test();

    # read LDAP settings
    Progress->NextStage();
    SambaBackend->Read();
#    if(Abort()) return false;
    
    # read firewall setting
    Progress->NextStage();
# HELPME:    SuSEFirewall->Read();
#    if(Abort()) return false;

    # Read finished
    Progress->NextStage();
    $Modified = 0;
    
    return 1;
}

BEGIN{ $TYPEINFO{Configured} = ["function", "boolean"] }
sub Configured {
    my ($self) = @_;

    # check /etc/samba/smb.conf
    return 0 unless SambaConfig->Configured();
    
    # check file /$VARDIR/samba_server_done_once
    my $stat = SCR->Read(".target.stat", DONE_ONCE_FILE);
    return 1 if defined $stat->{size};

    # check if the main config file is modified already
    my $res = SCR->Execute(".target.bash_output", "rpm -V samba-client | grep '/etc/samba/smb\.conf'");
    return 1 if $res && !$res->{"exit"} and $res->{"stdout"};

    return 0;
}


# Write all samba-server settings
# @param write_only if true write only
# @return true on success
BEGIN{ $TYPEINFO{Write} = ["function", "boolean", "boolean"] }
sub Write {
    my ($self, $write_only) = @_;

    # Samba-server read dialog caption
    my $caption = __("Saving Samba Server Configuration");

    # We do not set help text here, because it was set outside
    Progress->New($caption, " ", 4, [
	    # translators: write progress stage
	    _("Write the settings"),
	    # translators: write progress stage
	    _("Run SuSEconfig"),
	    # translators: write progress stage
	    ( !SambaService->GetServiceAutoStart() ? _("Disable Samba services") 
	    # translators: write progress stage
		: _("Enable Samba services") ),
	    # translators: write progress stage
	    _("Save firewall settings")
	], [
	    # translators: write progress step
	    _("Writing the settings..."),
	    # translators: write progress step
	    _("Running SuSEconfig..."),
	    # translators: write progress step
	    ! SambaService->GetServiceAutoStart() ? _("Disabling Samba services...") 
	    # translators: write progress step
		: _("Enabling Samba services..."),
	    # translators: write progress step
	    _("Saving firewall settings..."),
	    # translators: write progress step
	    _("Finished")
	],
	""
    );

    # if nothing to write, quit (but show at least the progress bar :-)
    Progress->NextStage();
    return 1 unless $self->GetModified();

    # check, if we need samba-pdb package
    my %backends = map {/:/;$`||$_,1} split " ", SambaConfig->GlobalGetStr("passdb backend", "");
    if($backends{mysql}) {
	PackageSystem->CheckAndInstallPackagesInteractive(["samba-pdb"]) or return 0;
    }

    # write settings
#    if (Abort()) return false;

    if (SambaConfig->Write($write_only)) {
    	Report->Error(__("Cannot write settings to /etc/samba/smb.conf."));
	return 0;
    }
    SCR->Execute(".target.bash", "touch " . DONE_ONCE_FILE);
    
    # run SuSEconfig for samba
#    if(Abort()) return false;
#    Progress->NextStage() unless $noprogress;

    SambaBackend->Write();    
    SambaService->Write();
    SambaTrustDom->Write();

#    if(Abort()) return false;

    # save firewall settings
#HELPME    SuSEFirewall->Write();
#    if(Abort()) return false;
    
    # progress finished
#    Progress->NextStage();

#    if(Abort()) return false;

    $GlobalsConfigured = 1;
    $Modified = 0;

    return 1;
}

# Get all samba-server settings from the first parameter
# (For use by autoinstallation.)
# @param settings The YCP structure to be imported.
BEGIN{ $TYPEINFO{Import} = ["function", "void", ["map", "any", "any"]] }
sub Import {
    my ($self, $settings) = @_;

    if ($settings and $settings->{"config"} and keys %{$settings->{"config"}}) {
	$GlobalsConfigured = 1;
    } else {
	$GlobalsConfigured = 0;
    }
    $Modified = 0;
	
    y2debug("Importing: ", Dumper($settings));

    SambaConfig->Import($settings->{"config"});
    SambaService->Import($settings->{"service"});
    SambaTrustDom->Import($settings->{"trustdom"});
    SambaBackend->Import($settings->{"backend"});
}

# Dump the samba-server settings to a single map
# (For use by autoinstallation.)
# @return map Dumped settings (later acceptable by Import ())
BEGIN{ $TYPEINFO{Export} = ["function", "any"]}
sub Export {
    my ($self) = @_;

    $GlobalsConfigured = 1 if $self->GetModified();
    $Modified = 0;
    
    return {
	version =>	"2.10",
	config =>	SambaConfig->Export(),
	backend =>	SambaBackend->Export(),
	service =>	SambaService->Export(),
	trustdom =>	SambaTrustDom->Export(),
    };
}

# Create a textual summary and a list of unconfigured options
# @return summary of the current configuration
BEGIN { $TYPEINFO{Summary} = ["function", "string"] }
sub Summary {
    my ($self) = @_;
    
    # summary header
    my $summary = "";
    
    unless ($GlobalsConfigured) {
	$summary = Summary->AddLine($summary, Summary->NotConfigured());
	return $summary;
    }
    
    # summary item: configured workgroup/domain
    $summary = Summary->AddHeader($summary, __("Global Configuration"));
    
    $summary = Summary->AddLine($summary, sprintf(__("Workgroup or Domain: %s"), SambaConfig->GlobalGetStr("workgroup", "")));

    if (SambaService->GetServiceAutoStart()) {
        # summary item: selected role for the samba server
        $summary = Summary->AddLine($summary, sprintf(__("Role: %s"), SambaRole->GetRoleName()));
    } else {
        # summary item: status of the samba service
        $summary = Summary->AddLine($summary, __("Samba server is <i>disabled</i>"));
    }

    # summary heading: configured shares
    $summary = Summary->AddHeader($summary, __("Share Configuration"));

    my $shares = SambaConfig->GetShares();
    
    if (!$shares or $#$shares<0) {
        # summary item: no configured shares
        $summary = Summary->AddLine($summary, __("none"));
    } else {
	$summary = Summary->OpenList($summary);
    	foreach(@$shares) {
	    my $path = SambaConfig->ShareGetStr($_, "path", undef);
	    $summary = Summary->AddListItem($summary, $_ . ($path ? " (<i>$path</i>)" : ""));
	
	    my $comment = SambaConfig->ShareGetComment($_);
    	    $summary = Summary->AddLine($summary, $comment) if $comment;
	};
	$summary = Summary->CloseList($summary);
    }

    return $summary;
}

# Return required packages for auto-installation
# @return map of packages to be installed and to be removed
BEGIN{$TYPEINFO{AutoPackages}=["function",["map","string",["list","string"]]]}
sub AutoPackages {
    my ($self) = @_;
    return { install=> $RequiredPackages, remove => []};
}

8;

