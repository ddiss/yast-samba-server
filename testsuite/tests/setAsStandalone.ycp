{

// test for SambaServer::setAsStandalone()
// $Id$

    include "testsuite.ycp";

    TESTSUITE_INIT( [
        $[ "target" : $[ "tmpdir" : "/tmp" ] ],
        $[],
        $[
            "target" : $[
                "bash_output" : $[],
            ],
        ]
    ], nil );
    
    import "Samba";
    import "SambaServer";
    
    DUMP(SambaServer::role);
    DUMP(Samba::security);
    DUMP( sformat( "Domain logons: %1", SambaServer::domain_logons) );
    DUMP( sformat( "Domain master: %1", SambaServer::domain_master) );
    DUMP( sformat( "Local master: %1", SambaServer::local_master) );
    DUMP( sformat( "Preferred master: %1", SambaServer::preferred_master) );
    
    Samba::modified = false;
    TEST( ``( SambaServer::setAsStandalone() ), [], nil );

    DUMP(SambaServer::modified);
    DUMP(SambaServer::role);
    DUMP(Samba::security);

    DUMP( sformat( "Domain logons: %1", SambaServer::domain_logons) );
    DUMP( sformat( "Domain master: %1", SambaServer::domain_master) );
    DUMP( sformat( "Local master: %1", SambaServer::local_master) );
    DUMP( sformat( "Preferred master: %1", SambaServer::preferred_master) );
}
