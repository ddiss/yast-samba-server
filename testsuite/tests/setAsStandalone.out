Dump	`standalone
Dump	user
Dump	Domain logons: nil
Dump	Domain master: nil
Dump	Local master: nil
Dump	Preferred master: nil
Return	true
Dump	false
Dump	`standalone
Dump	user
Dump	Domain logons: false
Dump	Domain master: false
Dump	Local master: nil
Dump	Preferred master: nil
